frontend/src/todos.js

```diff
@@ -1,10 +1,37 @@
 import React, { useState, useEffect } from 'react';
 import './todos.css';
+import Modal from "./modal";
 import listiclelogoandname from "./images/listiclelogoandname.png";
 // import listiclename from "./images/listiclename.png";
 
 function Todos() {
   const [todos, setTodos] = useState([]);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [formData, setFormData] = useState({ title: "", description: "" });
+
+  const handleAddTodo = (todoData) => {
```

Comment:
Before you even send anything to the server it might be good for you to carry out validation so that you can give the user early feedback, and avoid an otherwise unnecessary roundtrip to the server.